File: languages/php/Bloock/Entity/IdentityV2/Method.php
<?php

namespace Bloock\Entity\IdentityV2;

class Method
{
    const IDEN3 = "IDEN3";
    const POLYGON_ID = "POLYGON_ID";

    public static function fromProto(\Bloock\Method $method): string
    {
        switch ($method) {
            case \Bloock\Method::IDEN3:
                return Method::IDEN3;
            case \Bloock\Method::POLYGON_ID:
                return Method::POLYGON_ID;
            default:
                return Method::POLYGON_ID;
        }
    }

    public static function toProto(string $method): int
    {
        switch ($method) {
            case Method::IDEN3:
                return \Bloock\Method::IDEN3;
            case Method::POLYGON_ID:
                return \Bloock\Method::POLYGON_ID;
            default:
                return \Bloock\Method::POLYGON_ID;
        }
    }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/Method.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;

public enum Method {
    IDEN3,
    POLYGONID;

    public IdentityEntitiesV2.Method toProto() {
        switch (this) {
            case IDEN3:
                return IdentityEntitiesV2.Method.IDEN3;
            case POLYGONID:
                return IdentityEntitiesV2.Method.POLYGON_ID;
            default:
                return IdentityEntitiesV2.Method.POLYGON_ID;
        }
    }
}
